bump etcd to 3.2.8

soltysh opened the pull request:
This bumps etcd and performs additional cleanups needed for our restoration to work properly.

liggitt commented:
looks like the v2http package was renamed and is failing tests and build

liggitt commented:
at least, the part we use for NewPeerHandler (https://github.com/coreos/etcd/compare/v3.2.1...v3.2.8#diff-d7a47eb75475dba263540eb7f8456e50L152)

liggitt commented:
the etcd bump commit is much bigger than the [corresponding diff](https://github.com/coreos/etcd/compare/v3.2.1...v3.2.8) upstream... why is that?

liggitt commented on hack/godep-restore.sh:
```diff
@@ -56,8 +56,7 @@ preload-remote "github.com/openshift" "origin" "github.com/openshift" "origin" #
 preload-remote "k8s.io" "kubernetes" "github.com/openshift" "kubernetes"
 preload-remote "github.com/docker" "distribution" "github.com/openshift" "docker-distribution"
 preload-remote "github.com/skynetservices" "skydns" "github.com/openshift" "skynetservices-skydns"
-preload-remote "github.com/coreos" "etcd" "github.com/openshift" "coreos-etcd"
```
@smarterclayton why did we have this?
@soltysh do we no longer have any upstream commits?

soltysh replied:
I think we had those some time ago, but I've double checked the last bump and it was using official version (no carry commits). That's why I've decided to remove it from here. We'll bring it back when needed.

soltysh commented:
> the etcd bump commit is much bigger than the corresponding diff upstream... why is that?

I'll double check, but I'm guessing we were missing some tests, or alike.

soltysh commented:
> the etcd bump commit is much bigger than the corresponding diff upstream... why is that?

I've double checked and the reasons the bump is bigger than the difference are as follows:

1. Updated tests, a lot of them. I'm guessing previous bump was done without `-t` flag, which is enabled by default when invoking `hack/godep-save.sh` which I've used here.
2. Removed packages (eg. `github.com/coreos/etcd/clientv3/namespace` and `github.com/coreos/etcd/clientv3/naming`), looks like we don't need those anymore. 
3. Generated files (eg. `github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.gw.go`) removed in this bump. 

The tests are green on it, but I'll leave that up to you to decide @smarterclayton @liggitt @deads2k

soltysh commented:
@liggitt I've added that no-op bump commit for 3.2.1. With that in the actual 3.2.8 bump is a subset of https://github.com/coreos/etcd/compare/v3.2.1...v3.2.8
I wonder though, why we vendor integration tests for etcd, any particular reason, @smarterclayton ?

smarterclayton commented:
We don't need them.


On Oct 11, 2017, at 1:12 PM, Maciej Szulik <notifications@github.com> wrote:

@liggitt <https://github.com/liggitt> I've added that no-op bump commit for
3.2.1. With that in the actual 3.2.8 bump is a subset of coreos/etcd@
v3.2.1...v3.2.8 <https://github.com/coreos/etcd/compare/v3.2.1...v3.2.8>
I wonder though, why we vendor integration tests for etcd, any particular
reason, @smarterclayton <https://github.com/smarterclayton> ?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/16730#issuecomment-335881056>,
or mute
the thread
<https://github.com/notifications/unsubscribe-auth/ABG_p0D1t5y0JclugFSKvPx5FazUtgf8ks5srPb8gaJpZM4Pw3WH>
.

liggitt commented:
/lgtm

soltysh commented:
> We don't need them.

I'll do one another sweep through Godeps and I'll try removing them, in that case.

liggitt commented:
> We don't need them.
> 
> I'll do one another sweep through Godeps and I'll try removing them, in that case.

was the content of Godeps.json in this PR not the result of `hack/save-godeps.sh`? If so, I wouldn't worry about it for this PR

soltysh commented:
> was the content of Godeps.json in this PR the result of hack/save-godeps.sh? If so, I wouldn't worry about it for this PR

Yes, that's why I think we somehow rely on them. But maybe we don't any more, I'm not sure how good godep is with figuring out those deps.
